### PR TITLE
EVG-17405: Remove default logger from project page

### PR DESCRIPTION
**Pull request by julianedwards**
EVG-17405: https://jira.mongodb.org/browse/EVG-17405

### Description 
We no longer want project admins to switch back to using the Evergreen (database) logger. The ability to set the project ref's default logger field via the UI is removed with this PR. I still kept the field as "undiscoverable" since it will likely be useful in the near-future as we transition to whatever ends up replacing Cedar buildlogger.

I created https://jira.mongodb.org/browse/EVG-17669 for removing the field entirely once we are done changing the logging backend.

### Testing 
I tested that the field is no longer there in staging.

### CODE REVIEW

File: service/project.go [hunk 1]
```diff
@@ -570,7 +569,6 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 	projectRef.BatchTime = responseRef.BatchTime
 	projectRef.Branch = responseRef.Branch
 	projectRef.Enabled = &responseRef.Enabled
-	projectRef.DefaultLogger = responseRef.DefaultLogger
```
[the remark this answers is not in the document]
Comment (julianedwards) [reply service/project.go:573]: Ah, good catch! Thanks

**Comment by julianedwards** [comment]
> Can we get rid of
> 
> https://github.com/evergreen-ci/evergreen/blob/f53997ee98249519b99672fe78ee50b50d15154b/model/project_ref.go#L871
> 
> now? It seems like where Default logger is used (possibly just
> https://github.com/evergreen-ci/evergreen/blob/83945041f84a60db700e61e3b1fd09c08860ec37/agent/logging.go#L143
> 
> now that it's not in the UI) we can just set the default there if it's empty.

We still have a global admin setting for a default logger (exposed via the UI), so this sets it dynamically based on that. It is useful if we want to switch back to another logger if something goes wrong.

**Comment by ablack12** [comment]
Also should we remove from APIProjectRef?

**Comment by julianedwards** [comment]
> But wouldn't it be more effective to just look up the default at the time that we need it, instead of every single time we use project refs? (That seems more prone to error and loops through project unnecessarily; I think this was intentionally done to support showing it correctly on the UI / API but isn't relevant anymore)

This is a good point, I updated it to remove the `checkDefaultLogger` and only dynamically set it when the project ref is getting fetched by the agent. I also removed the field from `APIProjectRef`. I tested that this works in staging with a different logger than the global default and no default logger set.

**Review by ablack12 — APPROVED** [review]
Awesome 🚀

**Pull request merged**